File: src/main/java/net/dashflight/data/postgres/DashflightPostgresClientModule.java
package net.dashflight.data.postgres;

import com.google.inject.AbstractModule;

public class DashflightPostgresClientModule extends AbstractModule {

    @Override
    protected void configure() {
        bind(PostgresConnectionOptionProvider.class).to(DashflightPostgresConnectionOptionProvider.class);
    }

}

File: src/main/java/net/dashflight/data/postgres/DashflightPostgresClientModule.kt
package net.dashflight.data.postgres

import com.google.inject.AbstractModule

class DashflightPostgresClientModule constructor(private val applicationName: String? = null): AbstractModule() {

    override fun configure() {
        bind(PostgresConnectionOptionProvider::class.java).toInstance(DashflightPostgresConnectionOptionProvider(applicationName))
    }
}